yarn-project/aztec.js/src/fee/account_entrypoint_meta_payment_method.ts

```diff
@@ -0,0 +1,81 @@
+import {
+  EncodedAppEntrypointCalls,
+  EncodedCallsForEntrypoint,
+  computeCombinedPayloadHash,
+} from '@aztec/entrypoints/encoding';
+import type { AuthWitnessProvider, FeePaymentMethod } from '@aztec/entrypoints/interfaces';
+import { ExecutionPayload } from '@aztec/entrypoints/payload';
+import {
+  type ContractArtifact,
+  type FunctionArtifact,
+  FunctionCall,
+  FunctionSelector,
+  encodeArguments,
+  getFunctionArtifactByName,
+} from '@aztec/stdlib/abi';
+import { AztecAddress } from '@aztec/stdlib/aztec-address';
+import type { GasSettings } from '@aztec/stdlib/gas';
+
+/**
+ * Fee payment method that allows a contract to pay for its own deployment
+ * It works by rerouting the provided fee payment method through the account's entrypoint,
+ * which sets itself as fee payer.
+ */
+export class AccountEntrypointMetaPaymentMethod implements FeePaymentMethod {
+  constructor(
+    private artifact: ContractArtifact,
+    private authWitnessProvider: AuthWitnessProvider,
+    private feePaymentNameOrArtifact: string | FunctionArtifact,
+    private address: AztecAddress,
```

Comment:
```suggestion
    private accountAddress: AztecAddress,
```
Think this makes it clearer.